spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -79,33 +79,38 @@ public MessageChannel resolveDestination(String destinationName) {
 		synchronized (this) {
 			if (this.beanFactory != null && this.binderFactory != null) {
 				String[] dynamicDestinations = null;
-				Properties producerProperties = null;
 				if (this.channelBindingServiceProperties != null) {
 					dynamicDestinations = this.channelBindingServiceProperties.getDynamicDestinations();
-					// TODO: need the props to return some defaults if not found
-					producerProperties = this.channelBindingServiceProperties.getProducerProperties(destinationName);
 				}
 				boolean dynamicAllowed = ObjectUtils.isEmpty(dynamicDestinations)
-						|| ObjectUtils.containsElement(dynamicDestinations, destinationName);
+						|| ObjectUtils.containsElement(dynamicDestinations, channelName);
 				if (dynamicAllowed) {
-					String transport = null;
-					String beanName = destinationName;
-					if (destinationName.contains(":")) {
-						String[] tokens = destinationName.split(":", 2);
+					String binderName = null;
+					String beanName = channelName;
+					if (channelName.contains(":")) {
+						String[] tokens = channelName.split(":", 2);
 						if (tokens.length == 2) {
-							transport = tokens[0];
-							destinationName = tokens[1];
+							binderName = tokens[0];
+							channelName = tokens[1];
 						}
 						else if (tokens.length != 1) {
-							throw new IllegalArgumentException("Unrecognized channel naming scheme: " + destinationName + " , should be" +
-									" [<transport>:]<destinationName>");
+							throw new IllegalArgumentException("Unrecognized channel naming scheme: " + channelName + " , should be" +
+									" [<transport>:]<channelName>");
 						}
 					}
 					channel = new DirectChannel();
 					this.beanFactory.registerSingleton(beanName, channel);
 					channel = (MessageChannel) this.beanFactory.initializeBean(channel, beanName);
-					Binder<MessageChannel> binder = binderFactory.getBinder(transport);
-					this.dynamicDestinationsBindable.addOutputBinding(beanName, binder.bindProducer(destinationName, channel, producerProperties));
+					@SuppressWarnings("unchecked")
+					Binder<MessageChannel, ?, ProducerProperties> binder =
```

Comment:
Let's address this after rebasing https://github.com/spring-cloud/spring-cloud-stream/pull/424